repos/system_upgrade/el7toel8/models/initramfs.py

```diff
@@ -0,0 +1,123 @@
+    """
+    topic = BootPrepTopic
+
+    name = fields.String()
+    """
+    Name of the dracut module that should be added (--add option of dracut)
+    when a initramfs is built.
+    """
+
+    module_path = fields.Nullable(fields.String(default=None))
+    """
+    module_path specifies dracut modules that are supposed to be copied
+
+    If the path is not set, the given name will just be activated. IOW,
+    if the dracut module is stored outside the /usr/lib/dracut/modules.d/
+    directory, set the absolute path to it, so leapp will manage it during
+    the upgrade to ensure the module will be added into the initramfs.
+
+    The module has to be stored on the local storage. In such a case, it is
+    recommended to store it into the 'files' directory of an actor generating
+    this object.
+
+    Note: It's expected to set the full path from the host POV. In case
+    of actions inside containers, the module is still copied from the HOST
+    into the container workspace.
+    """
+
+
+class UpgradeInitramfsTasks(Model):
+    """
+    Influence generating of the (leapp) upgrade initramfs
+
+    The upgrade initramfs is used during the crucial part of the upgrade,
+    in which the original rpms are upgraded, configuration of applications
+    are migrated, etc. To be able to boot into the leapp upgrade environment
+    correctly, it is expected all needed drivers, configuration files, ... are
+    included inside the upgrade initramfs. Produce this message with
+    expected content to influence the upgrade initramfs.
+
+    If some specific rpms or content is required to be able to build the
+    upgrade initramfs, see the <container-model>.
+
+    Note: The built initramfs is composed of stuff for the target system.
+    In example, if you are on RHEL 7 and plan the upgrade to RHEL 8, you need
+    to provide content (e.g. drivers, dracut modules) compatible with
+    RHEL 8 system.
+    """
+    topic = BootPrepTopic
+
+    include_files = fields.List(fields.String(), default=[])
+    """
+    List of files (cannonical filesystem paths) to include in the initramfs
+    """
+
+    include_dracut_modules = fields.List(fields.Model(DracutModule), default=[])
+    """
+    List of dracut modules that should be installed in the initramfs.
+
+    See the DracutModule model for more information.
+    """
+
+
+class TargetInitramfsTasks(UpgradeInitramfsTasks):
+    """
+    Analogy to UpgradeInitramfsTasks, but referring to the target initram disk.
+
+    Target initramfs is the one, that will be used to boot to your upgraded
+    system. If you want to ensure that you are able to boot into the target
+    (upgraded) system, it is possible you need to add same stuff as you added
+    into the upgrade initramfs.
+
+    If some specific rpms are required to be able to build the upgrade
+    initramfs, install these via the RpmTransactionTasks model.
+    """
+
+
+@deprecated(since='2020-11-11', message='Replaced by TargetInitramfsTasks.')
```

Comment:
TODO: These dates should point to next release date.

Reply:
> @pirat89 for some reason I cannot hit the "conflicting dracut modules" inhibitor now. Do you have an idea what could go wrong?

Let's sync over video call to check it. If we find a bug, I will try to cover it by tests. but it's weird.

Reply:
The next release date cannot be specified here as we cannot communicate what is the release date before the release. As well, the date can be changed/postponed. I think we can keep this date as it is (or update it inside the "new release" PR). We are specifying end of the protection window in the documentation: https://leapp.readthedocs.io/en/latest/el7toel8/deprecation.html#deprecated-functionality-in-the-el7toel8-repository  (and the official one as well) - so even if the "since" is pointing to an older date, we are clarifying the end of the protection in additional documentations.